### PR TITLE
fix(ios): use searchtextfield exposed in iOS 13 to set color

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-27754

### CODE REVIEW

File: iphone/Classes/TiUISearchBar.m [hunk 1]
```diff
@@ -127,14 +127,27 @@ - (void)setHintTextColor_:(id)value
   id hintText = [[self proxy] valueForUndefinedKey:@"hintText"] ?: @"";
 
   NSAttributedString *placeholder = [[NSAttributedString alloc] initWithString:[TiUtils stringValue:hintText] attributes:@{ NSForegroundColorAttributeName : [[TiUtils colorValue:value] _color] }];
-  [[UITextField appearanceWhenContainedInInstancesOfClasses:@ [[UISearchBar class]]] setAttributedPlaceholder:placeholder];
+  if ([TiUtils isIOSVersionOrGreater:@"13.0"]) {
+#if IS_SDK_IOS_13
+    TiThreadPerformOnMainThread(
+        ^{
+          UISearchTextField *textField = [[self searchBar] searchTextField];
```
Comment (janvennemann) [inline iphone/Classes/TiUISearchBar.m]:
```suggestion
          UISearchTextField *textField = self.searchBar.searchTextField;
```

File: iphone/Classes/TiUISearchBar.m [hunk 2]
```diff
@@ -127,14 +127,27 @@ - (void)setHintTextColor_:(id)value
...
+          [textField setAttributedPlaceholder:placeholder];
+        },
+        NO);
+#endif
+  } else {
+    [[UITextField appearanceWhenContainedInInstancesOfClasses:@ [[UISearchBar class]]] setAttributedPlaceholder:placeholder];
```
Comment (janvennemann) [inline iphone/Classes/TiUISearchBar.m]:
```suggestion
    [[UITextField appearanceWhenContainedInInstancesOfClasses:@[UISearchBar.class] setAttributedPlaceholder:placeholder];
```

File: iphone/Classes/TiUISearchBar.m [hunk 3]
```diff
@@ -127,14 +127,27 @@ - (void)setHintTextColor_:(id)value
...
+  }
   RELEASE_TO_NIL(placeholder);
 }
 
 - (void)setColor_:(id)value
 {
   if ([TiUtils isIOSVersionOrGreater:@"13.0"]) {
+#if IS_SDK_IOS_13
     [[[self searchBar] searchTextField] setTextColor:[[TiUtils colorValue:value] _color]];
```
Comment (janvennemann) [inline iphone/Classes/TiUISearchBar.m]:
```suggestion
    self.searchBar.searchTextField.textColor = [TiUtils colorValue:value]._color;
```

File: iphone/Classes/TiUISearchBar.m [hunk 4]
```diff
@@ -127,14 +127,27 @@ - (void)setHintTextColor_:(id)value
   id hintText = [[self proxy] valueForUndefinedKey:@"hintText"] ?: @"";
 
   NSAttributedString *placeholder = [[NSAttributedString alloc] initWithString:[TiUtils stringValue:hintText] attributes:@{ NSForegroundColorAttributeName : [[TiUtils colorValue:value] _color] }];
-  [[UITextField appearanceWhenContainedInInstancesOfClasses:@ [[UISearchBar class]]] setAttributedPlaceholder:placeholder];
+  if ([TiUtils isIOSVersionOrGreater:@"13.0"]) {
+#if IS_SDK_IOS_13
+    TiThreadPerformOnMainThread(
+        ^{
+          UISearchTextField *textField = [[self searchBar] searchTextField];
+          [textField setAttributedPlaceholder:placeholder];
```
Comment (janvennemann) [inline iphone/Classes/TiUISearchBar.m]:
```suggestion
          textField.attributedPlaceholder = placeholder;
```

**Review by janvennemann — CHANGES_REQUESTED** [review]
Just some notes about code formatting. Please use property accessor where possible ;)

**Comment by vijaysingh-axway** [comment]
@janvennemann Updated!

**Review by janvennemann — APPROVED** [review]

**Comment by nguyen8427** [comment]
hello, is it possible to put this fixed in 8.3 or 8.4 release?

**Review by janvennemann — APPROVED** [review]

**Comment by ssaddique** [comment]
FR partially passed. See JIRA ticket for more details.

**Comment by ssaddique** [comment]
**FR Passed**. See JIRA ticket for details. PR ready to be merged.

**Pull request merged**